lib/imagery/testsuite/test_imagery_sigsetfile.py

```diff
@@ -0,0 +1,547 @@
+        self.assertEqual(So.nclasses, 1)
+        I_NewSubSig(ctypes.byref(So), ctypes.byref(So.ClassSig[0]))
+        self.assertEqual(So.ClassSig[0].nsubclasses, 1)
+
+        # Fill sigset struct with data
+        So.title = String("Signature title")
+        So.bandrefs[0] = ctypes.create_string_buffer(tempname(252).encode())
+        So.ClassSig[0].used = 1
+        So.ClassSig[0].classnum = 2
+        So.ClassSig[0].title = String("1st class")
+        So.ClassSig[0].type = 1
+        So.ClassSig[0].SubSig[0].pi = 3.14
+        So.ClassSig[0].SubSig[0].means[0] = 42.42
+        So.ClassSig[0].SubSig[0].R[0][0] = 69.69
+
+        # Write signatures to file
+        p_new_sigfile = I_fopen_sigset_file_new(self.sig_name)
+        sig_stat = os.stat(self.sigfile_name)
+        self.assertTrue(stat.S_ISREG(sig_stat.st_mode))
+        I_WriteSigSet(p_new_sigfile, ctypes.byref(So))
+        self.libc.fclose(p_new_sigfile)
+
+        # Read back from signatures file
+        Sn = SigSet()
+        I_InitSigSet(ctypes.byref(So), 0)
+        p_old_sigfile = I_fopen_sigset_file_old(self.sig_name)
+        ret = I_ReadSigSet(p_old_sigfile, ctypes.byref(Sn))
+        self.assertEqual(ret, -1)
+
+        # SigSet does not have free function
+
+    def test_roundtrip_sigset_v1_two_bands(self):
+        """Test writing and reading back sigset (v1) with two bands"""
+
+        # Create signature struct
+        So = SigSet()
+        I_InitSigSet(ctypes.byref(So), 2)
+        self.assertEqual(So.nbands, 2)
+        I_NewClassSig(ctypes.byref(So))
+        self.assertEqual(So.nclasses, 1)
+        I_NewSubSig(ctypes.byref(So), ctypes.byref(So.ClassSig[0]))
+        self.assertEqual(So.ClassSig[0].nsubclasses, 1)
+
+        # Fill sigset struct with data
+        So.title = String("Signature title")
+        So.bandrefs[0] = ctypes.create_string_buffer(b"The_Doors")
+        So.bandrefs[1] = ctypes.create_string_buffer(b"The_Who")
+        So.ClassSig[0].used = 1
+        So.ClassSig[0].classnum = 2
+        So.ClassSig[0].title = String("1st class")
+        So.ClassSig[0].type = 1
+        So.ClassSig[0].SubSig[0].pi = 3.14
+        So.ClassSig[0].SubSig[0].means[0] = 42.42
+        So.ClassSig[0].SubSig[0].means[1] = 24.24
+        So.ClassSig[0].SubSig[0].R[0][0] = 69.69
+        So.ClassSig[0].SubSig[0].R[0][1] = 13.37
+        So.ClassSig[0].SubSig[0].R[1][0] = 13.37
+        So.ClassSig[0].SubSig[0].R[1][1] = 21.21
+
+        # Write signatures to file
+        p_new_sigfile = I_fopen_sigset_file_new(self.sig_name)
+        sig_stat = os.stat(self.sigfile_name)
+        self.assertTrue(stat.S_ISREG(sig_stat.st_mode))
+        I_WriteSigSet(p_new_sigfile, ctypes.byref(So))
+        self.libc.fclose(p_new_sigfile)
+
+        # Read back from signatures file
+        Sn = SigSet()
+        p_old_sigfile = I_fopen_sigset_file_old(self.sig_name)
+        ret = I_ReadSigSet(p_old_sigfile, ctypes.byref(Sn))
+        self.assertEqual(ret, 1)
```

Comment:
Running on Mac gives:
```
FAIL: test_roundtrip_sigset_v1_two_bands (__main__.SigSetFileTestCase)
Test writing and reading back sigset (v1) with two bands
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Volumes/dev/grass/lib/imagery/testsuite/test_imagery_sigsetfile.py", line 194, in test_roundtrip_sigset_v1_two_bands
    self.assertEqual(ret, 1)
AssertionError: -1 != 1
```

Reply:
As there where memory handling related changes, please re-run the test.